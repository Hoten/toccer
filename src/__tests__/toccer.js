import toccer from '../'
import fs from 'fs'
import path from 'path'

const readFile = path => fs.readFileSync(path, 'utf8').replace(/\r\n/g, '\n')

describe('toccer', function () {
  var src = path.join(__dirname, 'examples')
  var files = fs.readdirSync(src).filter(file => fs.statSync(path.join(src, file)).isFile())

  files.forEach(function (file) {
    var input = readFile(path.join(src, file))
    var tocTemplate = readFile('./toc.mustache')

    var toccerized = toccer(input, tocTemplate)

    it(`example ${file} works`, function () {
      expect(toccerized).toMatchSnapshot()
    })

    it(`example ${file} is idempotent`, function () {
      expect(toccer(toccerized, tocTemplate)).toEqual(toccerized)
    })
  })
})
