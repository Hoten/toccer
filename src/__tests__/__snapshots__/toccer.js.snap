// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toccer example 1-hello.md works 1`] = `
"# Hello World!

[](toc)

*   [Section One](#section-one)
*   [Section Two](#section-two)
*   [Section Three](#section-three)

## Section One

content

## Section Two

content

## Section Three

content
"
`;

exports[`toccer example 2-subsections.md works 1`] = `
"# Hello World!

[](toc)

*   [Section One](#section-one)
*   [Section Two](#section-two)
    *   [Subsection One](#subsection-one)
    *   [Subsection Two](#subsection-two)
*   [Section Three](#section-three)

## Section One

content

## Section Two

### Subsection One

blah

### Subsection Two

blah

content

## Section Three

content
"
`;

exports[`toccer example 3-max-level.md works 1`] = `
"# Hello World!

[](toc&max-level=2)

*   [Section One](#section-one)
*   [Section Two](#section-two)
*   [Section Three](#section-three)

## Section One

content

## Section Two

### Subsection One

blah

### Subsection Two

blah

content

## Section Three

content
"
`;

exports[`toccer example 4-html-escapes.md works 1`] = `
"# Hello World!

[](toc)

*   [Section One](#section-one)
*   [Section Two](#section-two)
*   [Section Three](#section-three)

## Section One

> content

## Section Two

> content

## Section Three

> content
"
`;

exports[`toccer example 5-code-blocks.md works 1`] = `
"# Hello World!

[](toc)

*   [Section One](#section-one)

## Section One

\`\`\`javascript
var a = 100;
var b = 101;

if (a > b) {
  console.log('lies');
}
\`\`\`

\`\`\`
no language defined
oh well
\`\`\`
"
`;
